src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc'| 'desc';
 
+const getAverageGrades = (grades: number[]): number => {
+  return grades.length
+    ? grades.reduce((acc, cur) => acc + cur, 0) / grades.length
+    : 0;
+};
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents:Student[] = [...students];
+
+  return sortedStudents.sort((firstStud, secondStud) => {
+    const studAField = firstStud[sortBy];
+    const studBField = secondStud[sortBy];
```

Comment:
I guess it is more clear this way.
```suggestion
    const firstStudentProperty = firstStud[sortBy];
    const secondStudentProperty = secondStud[sortBy];
```